test-runner/src/fixtures.ts

```diff
@@ -152,24 +153,27 @@ export class FixturePool {
     return fn(params);
   }
 
-  async runTestWithFixtures(fn: Function, timeout: number, info: TestInfo) {
-    let timer: NodeJS.Timer;
-    const timerPromise = new Promise(f => timer = setTimeout(f, timeout));
+  async runTestWithFixturesAndTimeout(fn: Function, timeout: number, info: TestInfo) {
+    const { timedOut } = await raceAgainstTimeout(this._runTestWithFixtures(fn, info), timeout);
+    // Do not overwrite test failure upon timeout in fixture.
+    if (timedOut && info.result.status === 'passed')
+      info.result.status = 'timedOut';
+  }
+
+  async _runTestWithFixtures(fn: Function, info: TestInfo) {
+    try {
+      await this.resolveParametersAndRun(fn, info.config, info);
+      info.result.status = 'passed';
+    } catch (error) {
+      info.result.status = 'failed';
```

Comment:
`if (info.result.status === 'passed') info.result.status = 'failed';` to guard against failures after timeout changing the result.

Reply:
We are reporting immediately after the timeout, but sure.